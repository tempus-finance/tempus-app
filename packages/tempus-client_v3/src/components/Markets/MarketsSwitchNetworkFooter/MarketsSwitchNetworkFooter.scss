.tc__markets-switch-network-footer {
  height: 48px;
  background-color: var(--marketsFooterBackground);
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 0px 0px 40px 40px;

  .tc__actionButton {
    width: 96px;
    margin-left: 8px;
    border: none;
  }
}

.tc__markets-switch-network-footer__separator {
  width: 1px;
  height: 16px;
  margin: 0 4px;
  background-color: var(--marketsFooterSeparator);
}
