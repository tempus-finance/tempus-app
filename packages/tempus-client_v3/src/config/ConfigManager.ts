import { Chain, ChainConfig, Config, ProtocolName, TempusPool, Ticker } from 'tempus-core-services';
import config from './config';

interface TokenListItem {
  chain: Chain;
  address: string;
}

class ConfigManager {
  private config: Config = config;

  private chainList: Chain[] = [];
  private poolList: TempusPool[] = [];
  private tokenList: TokenListItem[] = [];

  init(): void {
    this.retrieveChainList();
    this.retrievePoolList();
    this.retrieveTokenList();
  }

  getConfig(): Config {
    return this.config;
  }

  getChainConfig(chain: Chain): ChainConfig {
    return this.config[chain];
  }

  getChainList(): Chain[] {
    return this.chainList;
  }

  getPoolList(): TempusPool[] {
    return this.poolList;
  }

  getTokenList(): TokenListItem[] {
    return this.tokenList;
  }

  // TODO add tests
  getFilteredPoolList(filterByChain?: Chain, filterByToken?: Ticker, filterByProtocol?: ProtocolName): TempusPool[] {
    return this.getPoolList()
      .filter(pool => {
        if (filterByChain) {
          return pool.chain === filterByChain;
        }

        return true;
      })
      .filter(pool => {
        if (filterByToken) {
          return pool.backingToken === filterByToken;
        }

        return true;
      })
      .filter(pool => {
        if (filterByProtocol) {
          return pool.protocol === filterByProtocol;
        }

        return true;
      });
  }

  getEarliestStartDate(filterByChain?: Chain, filterByToken?: Ticker, filterByProtocol?: ProtocolName): Date {
    const earliestPoolList = this.getPoolList()
      .filter(pool => {
        if (filterByChain) {
          return pool.chain === filterByChain;
        }

        return true;
      })
      .filter(pool => {
        if (filterByToken) {
          return pool.backingToken === filterByToken;
        }

        return true;
      })
      .filter(pool => {
        if (filterByProtocol) {
          return pool.protocol === filterByProtocol;
        }

        return true;
      })
      .sort((poolA, poolB) => (poolA.startDate < poolB.startDate ? -1 : 1));

    if (earliestPoolList && earliestPoolList.length) {
      return new Date(earliestPoolList[0].startDate);
    }

    throw new Error('getEarliestStartDate - Cannot find a pool by search criteria');
  }

  getMaturityDates(filterByChain?: Chain, filterByToken?: Ticker, filterByProtocol?: ProtocolName): Date[] {
    const today = new Date();
    const todayTime = today.getTime();

    const filteredPools = this.getFilteredPoolList(filterByChain, filterByToken, filterByProtocol)
      .filter(pool => todayTime <= pool.maturityDate)
      .sort((poolA, poolB) => (poolA.maturityDate < poolB.maturityDate ? -1 : 1));

    const dates: Date[] = [];
    filteredPools.forEach(pool => dates.push(new Date(pool.maturityDate)));

    return dates;
  }

  private retrieveChainList(): void {
    this.chainList = Object.keys(this.config) as Chain[];
  }

  private retrievePoolList(): void {
    const pools: TempusPool[] = [];

    Object.keys(this.config).forEach((chain: string) => {
      this.getChainConfig(chain as Chain).tempusPools.forEach((pool: TempusPool) => {
        pools.push({ ...pool, chain: chain as Chain });
      });
    });

    this.poolList = pools;
  }

  private retrieveTokenList(): void {
    const result: TokenListItem[] = [];

    const poolList = this.getPoolList();
    poolList.forEach(pool => {
      const poolChain = pool.chain;

      if (poolChain) {
        result.push(
          {
            address: pool.backingTokenAddress,
            chain: poolChain,
          },
          {
            address: pool.yieldBearingTokenAddress,
            chain: poolChain,
          },
          {
            address: pool.principalsAddress,
            chain: poolChain,
          },
          {
            address: pool.yieldsAddress,
            chain: poolChain,
          },
          {
            address: pool.ammAddress,
            chain: poolChain,
          },
        );
      } else {
        console.warn(`Pool ${pool.address} does not contain chain in it's config!`);
      }
    });

    this.tokenList = result;
  }
}

export default ConfigManager;
