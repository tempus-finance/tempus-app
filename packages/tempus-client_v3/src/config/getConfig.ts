import { Octokit } from 'octokit';
import { Config } from 'tempus-core-services';

const TempusPoolsConfig = {
  owner: 'tempus-finance',
  repo: 'tempus-pools-config',
  path: 'config.json',
};

let config: Config;
export const getConfig = async (): Promise<Config> => {
  if (!config) {
    try {
      const octokit = new Octokit();

      const response = await octokit.rest.repos.getContent({
        mediaType: {
          format: 'json',
        },
        ...TempusPoolsConfig,
      });

      if (response.status === 200) {
        const { data } = response;
        const { content } = data as any;

        if (content) {
          const decodedString = await String(Buffer.from(content, 'base64'));
          config = JSON.parse(decodedString);
        }
      }
    } catch (error) {
      console.log('Error retrieving Tempus Pools Config', error);
      return {};
    }
  }

  return config;
};
