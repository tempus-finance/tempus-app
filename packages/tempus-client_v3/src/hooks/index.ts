export { useTokenBalance, useTokenBalances } from './useTokenBalance';
export { usePoolBalance } from './usePoolBalance';
export { useUserDepositedPools } from './useUserDepositedPools';
export * from './useFixedAprs';
export * from './useTvlData';
export * from './useTokenRates';
export * from './useWalletAddress';
export * from './useSelectedChain';
export * from './useLocale';
export * from './useUserPreferences';
export * from './usePoolViewOptions';
export * from './useConfig';
export * from './useDepositModalData';
export * from './useYieldAtMaturity';
export * from './useChainList';
export * from './usePoolList';
export * from './useTokenList';
export * from './useServicesLoaded';
export * from './useApproveToken';
export * from './useSigner';
export * from './useAppEvent';
export * from './useAllowances';
