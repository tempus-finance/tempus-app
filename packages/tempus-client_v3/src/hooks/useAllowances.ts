import {
  BehaviorSubject,
  distinctUntilChanged,
  filter,
  merge,
  concatMap,
  Observable,
  withLatestFrom,
  of,
  from,
  map,
  Subscription,
  tap,
  debounce,
  scan,
  interval,
} from 'rxjs';
import { bind } from '@react-rxjs/core';
import { JsonRpcSigner } from '@ethersproject/providers';
import {
  Decimal,
  getDefinedServices,
  ZERO_ADDRESS,
  DEFAULT_DECIMAL_PRECISION,
  Config,
  ZERO,
} from 'tempus-core-services';
import { servicesLoaded$ } from './useServicesLoaded';
import { config$ } from './useConfig';
import { signer$ } from './useSigner';
import { walletAddress$ } from './useWalletAddress';
import { tokenApproveStatus$ } from './useApproveToken';
import { TokenBalance, tokenBalanceDataMap } from './useTokenBalance';
import { DEBOUNCE_IN_MS } from '../constants';

interface AllowanceMap {
  [chainTokenAddress: string]: {
    alwaysApproved: boolean;
    amount: Decimal;
  };
}

const DEFAULT_VALUE: AllowanceMap = {};

const rawAllowances$ = new BehaviorSubject<AllowanceMap>(DEFAULT_VALUE);
export const allowances$ = rawAllowances$.pipe(distinctUntilChanged());

const tokenBalance$ = merge(...Array.from(tokenBalanceDataMap.values()).map(({ subject$ }) => subject$));

// stream$ for listening token balance changes for fetching allowance
const tokenBalanceStream$ = tokenBalance$.pipe(
  withLatestFrom(config$, walletAddress$, signer$, servicesLoaded$),
  filter(
    ([, config, walletAddress, signer, servicesLoaded]) =>
      Boolean(config) && Boolean(walletAddress) && Boolean(signer) && servicesLoaded,
  ),
  concatMap<[TokenBalance, Config, string, JsonRpcSigner | null, boolean], Observable<AllowanceMap>>(
    ([{ balance, address, chain }, config, walletAddress, signer]) => {
      if (!balance || balance.lte(0) || address === ZERO_ADDRESS) {
        // always approved
        return of({
          [`${chain}-${address}`]: {
            alwaysApproved: true,
            amount: ZERO,
          },
        } as AllowanceMap);
      }

      try {
        const erc20TokenServiceGetter = getDefinedServices(chain).ERC20TokenServiceGetter(
          address,
          chain,
          signer as JsonRpcSigner,
        );
        const spenderAddress = config[chain].tempusControllerContract;
        return from(erc20TokenServiceGetter.getAllowance(walletAddress, spenderAddress)).pipe(
          map(
            bigNum =>
              ({
                [`${chain}-${address}`]: {
                  alwaysApproved: false,
                  amount: new Decimal(bigNum, DEFAULT_DECIMAL_PRECISION),
                },
              } as AllowanceMap),
          ),
        );
      } catch (error) {
        console.error(`useAllowance - Fail to get the allowance for token ${address} on ${chain}`, error);
        return of(DEFAULT_VALUE);
      }
    },
  ),
);

// stream$ for listening approval status
const approvalStream$ = tokenApproveStatus$.pipe(
  filter(status => Boolean(status.request) && status.success),
  map(
    ({ request }) =>
      ({
        [`${request?.chain}-${request?.tokenAddress}`]: {
          alwaysApproved: false,
          amount: request?.amount as Decimal,
        },
      } as AllowanceMap),
  ),
);

// merge all stream$ into one
const stream$ = merge(tokenBalanceStream$, approvalStream$).pipe(
  scan(
    (allAllowances, allowance) => ({
      ...allAllowances,
      ...allowance,
    }),
    {} as AllowanceMap,
  ),
  debounce<AllowanceMap>(() => interval(DEBOUNCE_IN_MS)),
  tap(poolTvls => rawAllowances$.next(poolTvls)),
);

export const [useAllowances] = bind(stream$, DEFAULT_VALUE);

let subscription: Subscription;

export const subscribeAllowance = (): void => {
  unsubscribeAllowance();
  subscription = stream$.subscribe();
};
export const unsubscribeAllowance = (): void => subscription?.unsubscribe?.();
export const resetAllowance = (): void => rawAllowances$.next(DEFAULT_VALUE);
