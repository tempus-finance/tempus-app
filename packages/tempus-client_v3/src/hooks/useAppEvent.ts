import { state, useStateObservable } from '@react-rxjs/core';
import { createSignal } from '@react-rxjs/utils';
import { TempusPool } from 'tempus-core-services';

export type AppEventType = 'deposit' | 'withdraw';

export interface AppEvent {
  eventType: AppEventType;
  tempusPool: TempusPool;
  txnHash: string;
  timestamp: number;
}

const [appEvent$, setAppEvent] = createSignal<AppEvent>();
const stateAppEvent$ = state(appEvent$, null);
export const emitAppEvent = setAppEvent; // to remind to use for better naming only

export function useAppEvent(): [AppEvent | null, (value: AppEvent) => void] {
  const appEvent = useStateObservable(stateAppEvent$);
  return [appEvent, emitAppEvent];
}

export { appEvent$ };
