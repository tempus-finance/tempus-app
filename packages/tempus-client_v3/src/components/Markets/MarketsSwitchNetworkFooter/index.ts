import MarketsSwitchNetworkFooter from './MarketsSwitchNetworkFooter';

export default MarketsSwitchNetworkFooter;
