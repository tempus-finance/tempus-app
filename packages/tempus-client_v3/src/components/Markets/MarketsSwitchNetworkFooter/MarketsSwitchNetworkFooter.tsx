import { useSetChain } from '@web3-onboard/react';
import { FC, useCallback, useMemo, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { chainNameToHexChainId, prettifyChainName } from 'tempus-core-services';
import { getConfigManager } from '../../../config/getConfigManager';
import { useSelectedChain } from '../../../hooks';
import ChainSelector from '../../ChainSelector';
import { ActionButton, Typography } from '../../shared';

import './MarketsSwitchNetworkFooter.scss';

const MarketsSwitchNetworkFooter: FC = () => {
  const chainList = getConfigManager().getChainList();
  const [, setChain] = useSetChain();
  const [selectedChain] = useSelectedChain();
  const { t } = useTranslation();

  const [chainSelectorOpen, setChainSelectorOpen] = useState<boolean>(false);

  const otherChains = useMemo(() => chainList.filter(chain => chain !== selectedChain), [chainList, selectedChain]);

  const title = useMemo(() => {
    if (otherChains.length === 1) {
      const chainName = prettifyChainName(otherChains[0]);
      return t('MarketsSwitchNetworkFooter.titleSingleChain', { chainName });
    }

    return t('MarketsSwitchNetworkFooter.titleMultipleChains');
  }, [otherChains, t]);

  const description = useMemo(() => {
    if (otherChains.length === 1) {
      const chainName = prettifyChainName(otherChains[0]);
      return t('MarketsSwitchNetworkFooter.descriptionSingleChain', { chainName });
    }

    return t('MarketsSwitchNetworkFooter.descriptionMultipleChains');
  }, [otherChains, t]);

  const handleSwitchClick = useCallback(() => {
    if (otherChains.length === 1) {
      const hexChainId = chainNameToHexChainId(otherChains[0]);

      if (hexChainId) {
        setChain({
          chainId: hexChainId,
        });
      }
    } else {
      setChainSelectorOpen(true);
    }
  }, [otherChains, setChain]);

  const onCloseChainSelector = useCallback(() => {
    setChainSelectorOpen(false);
  }, []);

  return selectedChain ? (
    <>
      <ChainSelector open={chainSelectorOpen} onClose={onCloseChainSelector} />
      <div className="tc__markets-switch-network-footer">
        <Typography variant="body-primary" weight="bold" color="text-disabled">
          {title}
        </Typography>
        <div className="tc__markets-switch-network-footer__separator" />
        <Typography variant="body-primary" color="text-disabled">
          {description}
        </Typography>
        <ActionButton variant="secondary" labels={{ default: 'Switch' }} onClick={handleSwitchClick} />
      </div>
    </>
  ) : null;
};

export default MarketsSwitchNetworkFooter;
