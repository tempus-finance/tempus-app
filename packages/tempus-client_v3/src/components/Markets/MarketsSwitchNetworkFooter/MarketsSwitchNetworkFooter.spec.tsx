import { fireEvent, render } from '@testing-library/react';
import { useSetChain } from '@web3-onboard/react';
import { chainNameToHexChainId } from 'tempus-core-services';
import { getConfigManager } from '../../../config/getConfigManager';
import { useSelectedChain } from '../../../hooks';
import MarketsSwitchNetworkFooter from './MarketsSwitchNetworkFooter';

const mockSetChain = jest.fn<void, [{ chainId: string }]>();

const subject = () => render(<MarketsSwitchNetworkFooter />);

jest.mock('@web3-onboard/react', () => ({
  useSetChain: jest.fn(),
}));

jest.mock('../../../hooks', () => ({
  ...jest.requireActual('../../../hooks'),
  useSelectedChain: jest.fn(),
}));

describe('MarketsSwitchNetworkFooter', () => {
  it('does not render footer if a chain is not selected', () => {
    (useSetChain as jest.Mock).mockReturnValue([{ connectedChain: null }, mockSetChain]);
    (useSelectedChain as jest.Mock).mockReturnValue([null]);

    const { container } = subject();

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('renders multi-chain footer if Ethereum chain is selected', () => {
    (useSetChain as jest.Mock).mockReturnValue([{ connectedChain: chainNameToHexChainId('ethereum') }, mockSetChain]);
    (useSelectedChain as jest.Mock).mockReturnValue(['ethereum', () => {}]);

    const { container } = subject();

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('opens chain selector when multi-chain footer is presented', () => {
    (useSetChain as jest.Mock).mockReturnValue([{ connectedChain: chainNameToHexChainId('ethereum') }, mockSetChain]);
    (useSelectedChain as jest.Mock).mockReturnValue(['ethereum', () => {}]);

    const { container, getByRole } = subject();

    const switchButton = getByRole('button');

    expect(container).not.toBeNull();
    expect(switchButton).not.toBeNull();

    fireEvent.click(switchButton);

    const chainSelector = container.querySelector('.tc__chainSelector .tc__modal');

    expect(chainSelector).not.toBeNull();
  });

  it('renders Fantom footer if Ethereum chain is selected', () => {
    (useSetChain as jest.Mock).mockReturnValue([{ connectedChain: chainNameToHexChainId('ethereum') }, mockSetChain]);
    (useSelectedChain as jest.Mock).mockReturnValue(['ethereum', () => {}]);
    jest.spyOn(getConfigManager(), 'getChainList').mockReturnValue(['ethereum', 'fantom']);

    const { container } = subject();

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('switches to Fantom if Ethereum chain is selected', () => {
    (useSetChain as jest.Mock).mockReturnValue([{ connectedChain: chainNameToHexChainId('ethereum') }, mockSetChain]);
    (useSelectedChain as jest.Mock).mockReturnValue(['ethereum', () => {}]);
    jest.spyOn(getConfigManager(), 'getChainList').mockReturnValue(['ethereum', 'fantom']);

    const { getByRole } = subject();

    const switchButton = getByRole('button');

    expect(switchButton).not.toBeNull();

    fireEvent.click(switchButton);

    expect(mockSetChain).toHaveBeenCalledTimes(1);
    expect(mockSetChain).toHaveBeenCalledWith({ chainId: chainNameToHexChainId('fantom') });
  });

  it('renders Ethereum footer if Fantom chain is selected', () => {
    (useSetChain as jest.Mock).mockReturnValue([{ connectedChain: chainNameToHexChainId('fantom') }, mockSetChain]);
    (useSelectedChain as jest.Mock).mockReturnValue(['fantom', () => {}]);
    jest.spyOn(getConfigManager(), 'getChainList').mockReturnValue(['ethereum', 'fantom']);

    const { container } = subject();

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('switches to Ethereum if Fantom chain is selected', () => {
    (useSetChain as jest.Mock).mockReturnValue([{ connectedChain: chainNameToHexChainId('fantom') }, mockSetChain]);
    (useSelectedChain as jest.Mock).mockReturnValue(['fantom', () => {}]);
    jest.spyOn(getConfigManager(), 'getChainList').mockReturnValue(['ethereum', 'fantom']);

    const { getByRole } = subject();

    const switchButton = getByRole('button');

    expect(switchButton).not.toBeNull();

    fireEvent.click(switchButton);

    expect(mockSetChain).toHaveBeenCalledTimes(1);
    expect(mockSetChain).toHaveBeenCalledWith({ chainId: chainNameToHexChainId('ethereum') });
  });
});
