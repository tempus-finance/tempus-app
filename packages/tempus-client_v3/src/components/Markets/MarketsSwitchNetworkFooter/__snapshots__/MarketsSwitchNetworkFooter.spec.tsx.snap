// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketsSwitchNetworkFooter does not render footer if a chain is not selected 1`] = `<div />`;

exports[`MarketsSwitchNetworkFooter renders Ethereum footer if Fantom chain is selected 1`] = `
<div>
  <div
    class="tc__chainSelector"
  />
  <div
    class="tc__markets-switch-network-footer"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(122, 122, 122); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Check out our Ethereum pools
    </div>
    <div
      class="tc__markets-switch-network-footer__separator"
    />
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(122, 122, 122); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Switch to the Ethereum network to view them
    </div>
    <button
      class="tc__btn tc__actionButton tc__actionButton-border-secondary-small tc__actionButton-background-secondary-small tc__actionButton-hover-secondary-small tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-small "
      data-selected="false"
      type="button"
    >
      <div
        class="tc__actionButton-status-label"
      >
        <div
          style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          Switch
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`MarketsSwitchNetworkFooter renders Fantom footer if Ethereum chain is selected 1`] = `
<div>
  <div
    class="tc__chainSelector"
  />
  <div
    class="tc__markets-switch-network-footer"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(122, 122, 122); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Check out our Fantom pools
    </div>
    <div
      class="tc__markets-switch-network-footer__separator"
    />
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(122, 122, 122); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Switch to the Fantom network to view them
    </div>
    <button
      class="tc__btn tc__actionButton tc__actionButton-border-secondary-small tc__actionButton-background-secondary-small tc__actionButton-hover-secondary-small tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-small "
      data-selected="false"
      type="button"
    >
      <div
        class="tc__actionButton-status-label"
      >
        <div
          style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          Switch
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`MarketsSwitchNetworkFooter renders multi-chain footer if Ethereum chain is selected 1`] = `
<div>
  <div
    class="tc__chainSelector"
  />
  <div
    class="tc__markets-switch-network-footer"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(122, 122, 122); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Check out our other pools
    </div>
    <div
      class="tc__markets-switch-network-footer__separator"
    />
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(122, 122, 122); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Switch to another network to view them
    </div>
    <button
      class="tc__btn tc__actionButton tc__actionButton-border-secondary-small tc__actionButton-background-secondary-small tc__actionButton-hover-secondary-small tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-small "
      data-selected="false"
      type="button"
    >
      <div
        class="tc__actionButton-status-label"
      >
        <div
          style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          Switch
        </div>
      </div>
    </button>
  </div>
</div>
`;
